Editor/AGS.Editor/Resources/agsdefns.sh

```diff
@@ -2471,6 +2529,36 @@ builtin struct Screen {
 };
 #endif
 
+#ifdef SCRIPT_API_v399
+builtin managed struct Joystick {
+  /// get number of connected joysticks.
+  import static readonly attribute int JoystickCount; // $AUTOCOMPLETESTATICONLY$
+  /// get a connected joystick by index or null on invalid index.
+  import static readonly attribute Joystick *Joysticks[]; // $AUTOCOMPLETESTATICONLY$
+  /// gets joystick or gamepad name
+  import readonly attribute String Name;
+  /// checks if joystick is really connected
+  import readonly attribute bool IsConnected;
+  /// checks if joystick is a valid gamepad
+  import readonly attribute bool IsGamepad;
+  /// checks if a gamepad button is pressed, including dpad.
+  import bool IsGamepadButtonDown(eGamepad_Button button);
+  /// get gamepad axis or trigger, trigger only has positive values.
+  import float GetGamepadAxis(eGamepad_Axis axis, float dead_zone = AXIS_DEFAULT_DEADZONE); 
+  /// get joystick axis or trigger, by index
+  import float GetAxis(int axis, float dead_zone = AXIS_DEFAULT_DEADZONE);
+  /// checks if a gamepad button is pressed, including dpad.
+  import bool IsButtonDown(int button);
+  /// checks if a gamepad button is pressed, including dpad.
+  import eJoystick_Hat GetHat(int hat);

```

Comment:
You seem to copy/pasted the function description for IsGamepadButtonDown, IsButtonDown and GetHat, these descriptions all say the same. Is this intended or a mistake?

In terms of style, all the existing comments to functions and properties in the script header begin with a capital letter, so you might change that for the style consistency too.

Reply:
Need to fix the comments, it's a mistake

Reply:
adjusted these comments and a few others